Spec readme role ats banjax swabber

kheops2713 opened the pull request:
This addresses #5.

I am not entirely sure about the software versions that we should put as default. 7.1.1 is our current in-production most modern version of ATS. The two other ones, I took them looking at the github repositories and using the most recent release version number. We could also specify latest on master branch, or a given commit ID. Proper version numbers seem cleaner to me though, but requires that banjax and swabber are correctly versioned.

kheops2713 committed:
Add README.md stub with structure, for ats-banjax-swabber role

Signed-off-by: KheOps <kheops@ceops.eu>

kheops2713 committed:
ats-banjax-swabber role's README.md: add state description regarding
software installation

Signed-off-by: KheOps <kheops@ceops.eu>

kheops2713 committed:
Add ATS, Banjax and swabber versions as input data for the role, and
links to Banjax and swabber source code repositories.

Signed-off-by: KheOps <kheops@ceops.eu>

kheops2713 committed:
Add specification that ATS, Banjax and swabber compilation-only dependencies are not kept

Signed-off-by: KheOps <kheops@ceops.eu>

tomryanx reviewed (dismissed):
It's a little unclear to me where sources *will* live, and whether it's possible to use a package manager for wider deployment.

kheops2713 commented:
Do you agree with the default version numbers as they are now? I'd like an explicit confirmation (or denial) that Banjax and Swabber have been properly versioned, in order to put good defaults.

Sources will be pulled using `git` or downloading tarball depending on the case, into any temporary directory. They will then be used to compile but will other not "live" as they'll be removed from the target system post-compilation. Is this clearer?

tomryanx commented:
As ATS 7.1.1 is the current release, that seems very reasonable. I prefer this reason than a comparison to what we happen to run elsewhere (although that's also interesting). About banjax and swabber, I'm not at all familiar with their versions. I see that the latest release of swabber is indeed 1.3.5, but the latest commit in master is 1.3.7 - are you confident we shouldn't be using (potentially marking as a release) 1.3.7? For banjax, all I see is that the latest release is 2.0.0, so that looks good as is.

Potential packaging is still not clear to me. In particular, I'm wondering whether edges will be able (via ansible apt) to pull updated packages. From the sounds of it, it will be compiled either on every edge, or every time something is deployed. If compilation is done periodically on the controller, then care will need to be taken with install scripts and dependencies on the edge.

kheops2713 commented:
Ack for the versions, will double-check why I did not see 1.3.7 and bump it up. My intent was to put the latest I found.

In isolation, this role would compile and then install ATS+banjax+swabber from their sources all on a "target" system. Concretely this is aimed at an edge, yes. If there exists up-to-date enough packaged binaries of any of those, clearly it'll be easier than use the source. If there is, then the spec should be changed, but I am not aware of any? Am I missing something?

tomryanx commented:
I would favour a kinda D.R.Y. approach of having compiled artefacts of some sort to push to (or to be pulled from) edges, rather than compiling everything on all edges over and over again.

kheops2713 commented:
Agreed, it's better if compilation can happen just once on a non-edge system. I will modify this specification accordingly so that the role can also fetch binaries from a given location.

Also, I cannot find Swabber 1.3.7 there: https://github.com/equalitie/swabber/releases. I feel like some tagging is needed over there…

kheops2713 committed:
Change post-execution state description to match discussions: the
various pieces of software may also be fetched in binary form from
somewhere else

Signed-off-by: KheOps <kheops@ceops.eu>

kheops2713 committed:
Change ats-banjax-swabber role input data specification to support fetching binaries directly

Signed-off-by: KheOps <kheops@ceops.eu>

RamJett commented:
Tested and currently running on a test edge is Debian-9, ATS-7.1.1, Banjax-v2.0, Swabber-1.3.7. Swabber last updates made it compatible with Debian-9. Previous versions do not work with Debian-9


On 01/16/2018 05:59 AM, tomryanx wrote:
>
> As ATS 7.1.1 is the current release, that seems very reasonable. I
> prefer this reason than a comparison to what we happen to run
> elsewhere (although that's also interesting). About banjax and
> swabber, I'm not at all familiar with their versions. I see that the
> latest release of swabber is indeed 1.3.5, but the latest commit in
> master is 1.3.7 - are you confident we shouldn't be using (potentially
> marking as a release) 1.3.7? For banjax, all I see is that the latest
> release is 2.0.0, so that looks good as is.
>
> Potential packaging is still not clear to me. In particular, I'm
> wondering whether edges will be able (via ansible apt) to pull updated
> packages. From the sounds of it, it will be compiled either on every
> edge, or every time something is deployed. If compilation is done
> periodically on the controller, then care will need to be taken with
> install scripts and dependencies on the edge.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/equalitie/deflect/pull/11#issuecomment-357968680>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABvo64bWjJQXCZkO8_wm7DNwtYHvmG5dks5tLKtXgaJpZM4Reayn>.
>

kheops2713 commented:
> Tested and currently running on a test edge is Debian-9, ATS-7.1.1, Banjax-v2.0, Swabber-1.3.7. Swabber last updates made it compatible with Debian-9. Previous versions do not work with Debian-9

OK I spent a bit of time looking around at Swabber and the versioning it quite messed up. The latest git tag is 1.3.5 which is desynchronised from the latest version in `setup.py` that was set in 2015 to 1.3.7… Which is itself not up to date considering that there are much more recent changes in the master branch. From the outside, 1.3.5 is the only version that properly exists so it will stay this way until Swabber's tagging is fixed.

@tomryanx I will let you take a look at the new formulations regarding installing the software from binaries on the target system and see if that seems a good specification!

tomryanx commented:
On L25 and L27, should `default: none` be changed to `required field` or similar? Or does 'none' result in the package not being installed?

As to binary vs compile, it seems like there are 3 options:
1. get a binary package from somewhere, push to target(s) and install
1. get source, compile on the controller, push compiled artefacts to target(s) and install
1. get source, push to each target, compile locally and install

I think we now have the first and last options covered only. I still like the option of using source, but compiling only once. I don't think this could be considered a "specific case" of the first option, because the first option doesn't appear to compile anything.

I also feel like this is splitting some pretty small hairs, so I have gone ahead and approved - I think it's adequate to proceed in it's present state.
